content/learn/beginner/datatypes-1.md

```diff
@@ -0,0 +1,131 @@
+2. As a base-16 number, such as `0x2A`. This is the [hexadecimal representation][hex] of the number forty-two.
+3. As a base-8 number, such as `052`. This is the [octal representation][oct] of the number forty-two.
+4. As a base-2 number, such as `0b101010`. This is the [binary representation][bin] of the number forty-two.
+
+[hex]: https://en.wikipedia.org/wiki/Hexadecimal
+[oct]: https://en.wikipedia.org/wiki/Octal
+[bin]: https://en.wikipedia.org/wiki/Binary_number
+
+### Floating-Point Numbers
+
+We represent floating-point numbers as 64-bit IEEE-754 floating-point numbers. This means that you can store numbers
+with a precision of about 15 decimal places. If you try to store a number with more precision, the bot will round it to
+the nearest representable number.
+
+There are a lot of ways to define a floating-point number, but the most common way is to use the decimal point, such as
+`3.14`. For a full list of ways to define a floating-point number, you can refer to the
+[Go documentation](https://golang.org/ref/spec#Floating-point_literals).
+
+### Imaginary Numbers
+
+Go (the language the bot is written in) also supports imaginary numbers (and by extension complex numbers).
+
+Defining them is quite similar to the formats discussed above, but with the addition of the imaginary unit `i`.
+Please refer to the [Go documentation](https://golang.org/ref/spec#Imaginary_literals) for more information.
+
+## Booleans
+
+A boolean is a data type that can only have one of two values: `true` or `false`. Booleans are used to represent the
+truth value of an expression. For instance, `5 > 3` would evaluate to `true`, while `5 < 3` would evaluate to `false`.
+
+You can think of it as a light switch: it can either be on (`true`) or off (`false`). Booleans are used in conditional
+statements, such as `if` statements, to determine which branch of the code should be executed.
+
+## Variables
+
+All this talk about data and data types is nice, but how do we actually use them in our code and keep them around for
+later use? This is where variables come in. A variable is a way to store a value and give it a name. You can then refer
+to this value by its name later in your code.
+
+In Custom Commands, you can define a variable by starting with a `$`, its name, and the assignment operator `:=`. Take
+the following code snippet, for example:
+
+```go
+{{ $name := "Alice" }}
+{{ $age := 42 }}
+{{ $isCool := true }}
+
+{{ $name }}, aged {{ $age }}, is cool: {{ $isCool }}
+```
+
+Later on, we may wish to reassign a new value to a variable. We can do this by using the re-assignment operator `=`.
+When and why this becomes necessary will be discussed in a later chapter.
+
+When debugging your code, you might have to figure out the type of a certain variable. You can do this by using the
+`printf` function with the `%T` verb, like so:
+
+```go
+{{ $name := "Alice" }}
+{{ printf "%T" $name }}
+```
```

Comment:
If you're going to introduce `printf` here, either link to further documentation on it, or allude to discussing it more in depth in a later chapter. As it sits, this page feels incomplete, it left me wanting more.